dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -102,10 +102,12 @@ class TeacherApplication < ApplicationBase
     # based on these rules in order:
     # 1. Application has a specific school? always overwrite the user's school info
     # 2. User doesn't have a specific school? overwrite with the custom school info.
+    # Will only update the school info if we have enough information for it
+    # An incomplete application may not have all the information we need to update
     def update_user_school_info!
       if school_id || user.school_info.try(&:school).nil?
-        school_info = get_duplicate_school_info(school_info_attr) || SchoolInfo.create!(school_info_attr)
```

Comment:
If `school_info_attr` returns nil, this line fails. Instead, I only run `update_school_info` if `school_info_attr` returns something.

Reply:
Another option is to not do this until the application is complete––checking with Tess, who's checking with RED, to see what's preferred